### ISSUE TITLE
Maximum Update Depth

### ISSUE THREAD

**Issue by giovanicascaes** [benhurott/react-native-masked-text#125]
Hello!

I can't use the component on React Native version 0.55, I'm getting the following error: `Maximum update depth excedeed`. Tried with minimum options.

**Comment by benhurott** [comment]
Hi!

Could you add the code you used?

**Comment by giovanicascaes** [comment]
Unfortunately is a private repo, so I can't. But I using the component with React Native Paper's TextInput, like this:

```js
<PersonInfoInput
            theme={{
              colors: { background: colors.screenBackground }
            }}
            mode="outlined"
            label="CPF*"
            value={this.state.cpf}
            onChangeText={text => this.setState({ cpf: text })}
            disabled={this.state.destination === null}
            render={_props => <TextInputMask type="cpf" />}
          />
```

**Comment by benhurott** [comment]
Ok, you could have the issue inside the other part of the code, but this code you shared with me is wrong.
Why?
You are not passing the `value` and the `onChangeText` to the TextInputMask (my bad, I need to document this better). 

```
<TextInputMask
          ref={(ref) => this.myDateText = ref}
          // here we set the custom component and their props.
          customTextInput={Kaede}
          customTextInputProps={{
            style:{ width: '80%' },
            label:'Birthday'
          }}

          type={'datetime'}
          options={{
            format: 'DD-MM-YYYY HH:mm:ss'
          }}

          // don't forget: the value and state!
          onChangeText={birthday => this.setState({ birthday })}
          value={this.state.birthday} />
```

About the maximum update, it could be cause by a `setState loop` inside other render functions.

Well, if you can't share the code because it's private, can you create a new React-Native project and test the same scenario?
I will keep this issue opened for some time.

**Comment by giovanicascaes** [comment]
Hello! Thanks for the friendly reply.

I checked my code again and after a few hours found that It seems the masked component do not like too much Styled Components. Using native wrappers instead, like View and RN Paper's TextInput I can get my screen showing. I coded something like this:

```js
import React, { Component } from "react";
import { View } from "react-native";
import { TextInputMask } from "react-native-masked-text";

export default class extends Component {
  render() {
    return (
      // Other native wrappers
      <View>
        // Other native wrappers
        <TextInput render={props => <TextInputMask {...props} type="cpf" />} />
        // Closing native wrappers
      </View>
      // Closing native wrappers
    );
  }
}
```

However the masked component now is allowing CPF > 14 on iOS (on Android the value is reverted after about 1 sec). I will dig a little more, since `checkText` workaround didn't work. Sorry not showing a working exemple, but it seems Snack also do not like Style Components either currently and I don't have much time to setup a new project to test right now.

**Comment by benhurott** [comment]
@giovanicascaes yes, but this is an issue with React-Native and not with the library.
This issue was fixed in RN repository for most recent versions.
Look at this issue and you can fix: [iOS allows too many characters #68](https://github.com/benhurott/react-native-masked-text/issues/68)

I will close it, for now, feel free to reopen if need ok?

o/
